pkgs/applications/science/logic/rocq-core/default.nix

```diff
@@ -0,0 +1,152 @@
+      ; If the coq-library path was already set, re-set it based on our current rocq
+      (when (fboundp 'get-coq-library-directory)
+        (inherit-local-permanent coq-library-directory (get-coq-library-directory))
+        (coq-prog-args))
+      (mapc (lambda (arg)
+        (when (file-directory-p (concat arg "/lib/coq/${rocq-version}/user-contrib"))
+          (setenv "COQPATH" (concat (getenv "COQPATH") ":" arg "/lib/coq/${rocq-version}/user-contrib")))) '(${lib.concatStringsSep " " (map (pkg: "\"${pkg}\"") pkgs)}))
+      ; TODO Abstract this pattern from here and nixBufferBuilders.withPackages!
+      (defvar nixpkgs--rocq-buffer-count 0)
+      (when (eq nixpkgs--rocq-buffer-count 0)
+        (make-variable-buffer-local 'nixpkgs--is-nixpkgs-rocq-buffer)
+        (defun nixpkgs--rocq-inherit (buf)
+          (inherit-local-inherit-child buf)
+          (with-current-buffer buf
+            (setq nixpkgs--rocq-buffer-count (1+ nixpkgs--rocq-buffer-count))
+            (add-hook 'kill-buffer-hook 'nixpkgs--decrement-rocq-buffer-count nil t))
+          buf)
+        ; When generating a scomint buffer, do inherit-local inheritance and make it a nixpkgs-rocq buffer
+        (defun nixpkgs--around-scomint-make (orig &rest r)
+          (if nixpkgs--is-nixpkgs-rocq-buffer
+              (progn
+                (advice-add 'get-buffer-create :filter-return #'nixpkgs--rocq-inherit)
+                (apply orig r)
+                (advice-remove 'get-buffer-create #'nixpkgs--rocq-inherit))
+            (apply orig r)))
+        (advice-add 'scomint-make :around #'nixpkgs--around-scomint-make)
+        ; When we have no more rocq buffers, tear down the buffer handling
+        (defun nixpkgs--decrement-rocq-buffer-count ()
+          (setq nixpkgs--rocq-buffer-count (1- nixpkgs--rocq-buffer-count))
+          (when (eq nixpkgs--rocq-buffer-count 0)
+            (advice-remove 'scomint-make #'nixpkgs--around-scomint-make)
+            (fmakunbound 'nixpkgs--around-scomint-make)
+            (fmakunbound 'nixpkgs--rocq-inherit)
+            (fmakunbound 'nixpkgs--decrement-rocq-buffer-count))))
+      (setq nixpkgs--rocq-buffer-count (1+ nixpkgs--rocq-buffer-count))
+      (add-hook 'kill-buffer-hook 'nixpkgs--decrement-rocq-buffer-count nil t)
+      (setq nixpkgs--is-nixpkgs-rocq-buffer t)
+      (inherit-local 'nixpkgs--is-nixpkgs-rocq-buffer)
+    '';
+  };
+
+  nativeBuildInputs = [ pkg-config ]
+    ++ ocamlNativeBuildInputs;
+  buildInputs = [ ncurses ];
+
+  propagatedBuildInputs = ocamlPropagatedBuildInputs;
+
+  postPatch = ''
+    UNAME=$(type -tp uname)
+    RM=$(type -tp rm)
+    substituteInPlace tools/beautify-archive --replace-warn "/bin/rm" "$RM"
+    ${csdpPatch}
+  '';
+
+  setupHook = writeText "setupHook.sh" ''
+    addCoqPath () {
+      if test -d "''$1/lib/coq/${rocq-version}/user-contrib"; then
+        export COQPATH="''${COQPATH-}''${COQPATH:+:}''$1/lib/coq/${rocq-version}/user-contrib/"
+      fi
+    }
+
```

Comment:
Is it wise to use the `COQPATH` environment variable? What about using `ROCQPATH` instead?